fearless/Modules/Staking/Analytics/Model/AnalyticsPeriod.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+enum AnalyticsPeriod: CaseIterable {
+    case weekly
+    case monthly
+    case yearly
+}
+
+extension AnalyticsPeriod {
+    static let `default` = AnalyticsPeriod.monthly
+
+    func title(for _: Locale) -> String {
+        switch self {
+        case .weekly:
+            return "weekly".uppercased()
+        case .monthly:
+            return "monthly".uppercased()
+        case .yearly:
+            return "yearly".uppercased()
+        }
+    }
+
+    var chartBarsCount: Int {
+        xAxisValues.count
+    }
+
+    var xAxisValues: [String] {
+        switch self {
+        case .weekly:
+            return ["M", "T", "W", "T", "F", "S", "S"]
+        case .monthly:
+            return ["1", "7", "15", "22", "31"]
+        case .yearly:
+            return (1 ... 12).map { String($0) }
+        }
+    }
+}
+
+extension AnalyticsPeriod {
+    func timestampInterval(periodDelta: Int) -> (Int64, Int64) {
+        let tillDate: Date = {
+            let interval: TimeInterval = {
+                switch self {
+                case .weekly:
+                    return .secondsInDay * 7
+                case .monthly:
+                    return .secondsInDay * 31
+                case .yearly:
+                    return .secondsInDay * 31 * 12
+                }
+            }()
+            return Date().addingTimeInterval(interval * Double(periodDelta))
+        }()
+
+        let calendar = Calendar(identifier: .iso8601)
```

Comment:
I would suggest to pass calendar outside. Probably, store it in presenter and pass to view model factory. Then we also need to subscribe to day changes in presenter using this calendar (at least create task for this).